tests/test_models.py

```diff
@@ -1,40 +1,153 @@
+    """ Test Cases for Customer Model """
 
     @classmethod
     def setUpClass(cls):
         """ This runs once before the entire test suite """
-        pass
+        app.config["TESTING"] = True
+        app.config["DEBUG"] = False
+        app.config["SQLALCHEMY_DATABASE_URI"] = DATABASE_URI
+        app.logger.setLevel(logging.CRITICAL)
+        Customer.init_db(app)
 
     @classmethod
     def tearDownClass(cls):
         """ This runs once after the entire test suite """
-        pass
+        db.session.close()
 
     def setUp(self):
         """ This runs before each test """
-        pass
+        db.session.query(Customer).delete()  # clean up the last tests
+        db.session.commit()
 
     def tearDown(self):
         """ This runs after each test """
-        pass
+        db.session.remove()
 
     ######################################################################
     #  T E S T   C A S E S
     ######################################################################
 
-    def test_XXXX(self):
-        """ It should always be true """
-        self.assertTrue(True)
+    def test_create_a_customer(self):
+        """ It should Create an Customer and assert that it exists """
+        customer = Customer(name="Katerine",lastname="Perdomo",email="kate@email.com",phone="XXXXXX",address="casa",city="Bogota",state='CU',country="Colombia",zipcode="11023")
+        self.assertTrue(customer is not None)
+        self.assertEqual(customer.id, None)
+        self.assertEqual(customer.name, "Katerine")
+        self.assertEqual(customer.lastname, "Perdomo")
+        self.assertEqual(customer.email, "kate@email.com")
+        self.assertEqual(customer.phone, "XXXXXX")
+        self.assertEqual(customer.address, "casa")
+        self.assertEqual(customer.state, "CU")
+        self.assertEqual(customer.country, "Colombia")
+        self.assertEqual(customer.zipcode, "11023")
+
+    def test_add_a_customer(self):
+        """It should Create a customer and add it to the database"""
+        customers = Customer.all()
+        self.assertEqual(customers, [])
+        customer = Customer(name="Katerine",lastname="Perdomo",email="kate@email.com",phone="XXXXXX",address="casa",city="Bogota",state='CU',country="Colombia",zipcode="11023")
+        self.assertTrue(customer is not None)
+        self.assertEqual(customer.id, None)
+        customer.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(customer.id)
+        customers = Customer.all()
+        self.assertEqual(len(customers), 1)
+    
+    def test_read_a_customer(self):
+        """It should Read a Customer"""
+        customer = CustomerFactory()
+        logging.debug(customer)
+        customer.id = None
+        customer.create()
+        self.assertIsNotNone(customer.id)
+        # Fetch it back
+        found_customer = Customer.find(customer.id)
+        self.assertEqual(found_customer.id, found_customer.id)
```

Comment:
@mm12063 
I think this is right, because I only want to read the customer

Reply:
But the assert will always be true. You're testing for the same data, for example: 
```
firstname = 'mitch'
self.assertEqual(firstname, firstname) #Both of them will be 'mitch' so it'll alway be equal/true.
```

I think it should be: `self.assertEqual(customer.id, found_customer.id)` because you're trying to confirm that both the customer you create and then store in the db (and then retrieve/read it) have the same id. The same for the other customer data below.

Reply:
🤦🏽‍♀️ 
You're right